core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/worker/managers/OutputPortResultWriterThread.scala

```diff
@@ -0,0 +1,43 @@
+package edu.uci.ics.amber.engine.architecture.worker.managers
+
+import com.google.common.collect.Queues
+import edu.uci.ics.amber.core.storage.model.BufferedItemWriter
+import edu.uci.ics.amber.core.tuple.Tuple
+import edu.uci.ics.amber.core.workflow.PortIdentity
+
+import java.util.concurrent.CompletableFuture
+import scala.collection.mutable
+
+sealed trait TerminateSignal
+case object TerminateSignalInst extends TerminateSignal
+
+class OutputPortResultWriterThread(writer: BufferedItemWriter[Tuple]) extends Thread {
+
+  private val queue = Queues.newLinkedBlockingQueue[Either[Tuple, TerminateSignal]]()
+  private var stopped = false
+  private val gracefullyStopped = new CompletableFuture[Unit]()
+
+  def putTuple(tuple: Tuple): Unit = {
+    assert(!stopped)
+    queue.put(Left(tuple)) // Non-blocking enqueue
+  }
```

Comment:
same comment. I suggest expose a queue out instead of expose this method. It is strange to call a function on a thread.

Reply:
Done